src/tdsops.f90

```diff
@@ -88,9 +89,23 @@ function tdsops_init(tds_n, delta, operation, scheme, bc_start, bc_end, &
     logical, optional, intent(in) :: sym !! (==npaire), only for Neumann BCs
     real(dp), optional, intent(in) :: c_nu, nu0_nu !! params for hypervisc.
 
-    integer :: n_stencil
+    integer :: n, n_stencil
+
+    tdsops%n_tds = n_tds
 
-    tdsops%tds_n = tds_n
+    ! we need special treatment in the right-hand-side build stage for
+    ! the very last point in the domain if output length is smaller than
+    ! the input length
```

Comment:
To check my understanding: `n_tds` is the number of equations to solve, `n_rhs` is the total number of points in the RHS stencils?

Reply:
Yes, provided an explanation in the `tdsops_t` class where they're declared as class members